src/ansys/aedt/core/generic/quaternion.py

```diff
@@ -0,0 +1,928 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 - 2025 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import math
+
+from ansys.aedt.core.generic.general_methods import pyaedt_function_handler
+from ansys.aedt.core.generic.math_utils import MathUtils
+from ansys.aedt.core.modeler.geometry_operators import GeometryOperators
+
+
+class Quaternion:
+    """
+    Implements fundamental quaternion operations.
+
+    Quaternions are created using ``Quaternion(a, b, c, d)``.
+
+    Quaternions are only used to represent rotations in 3D space.
+    They are not used to represent translations or other transformations.
+    Only methods related to rotations are implemented.
+
+    The quaternion is defined as:
+    .. math::
+        q = a + bi + cj + dk
+    where ``a`` is the scalar part and ``b``, ``c``, and ``d`` are the vector parts.
+
+    This updated class offers enhanced functionality compared to the previous implementation,
+    supporting both intrinsic and extrinsic rotations. Note that AEDT coordinate systems use intrinsic rotation.
+
+
+    References
+    ----------
+
+    [1] https://en.wikipedia.org/wiki/Quaternion
+    [2] https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles
+    [3] https://www.euclideanspace.com/maths/geometry/rotations/conversions/
+
+    """
+
+    def __init__(self, a=0, b=0, c=0, d=0):
+        """Initialize the quaternion.
+        Quaternions are created using ``Quaternion(a, b, c, d)``, representing the form q = a + bi + cj + dk.
+
+        Parameters
+        ----------
+
+        a, b, c, d : float
+            The quaternion coefficients.
+        """
+        self._args = (a, b, c, d)
```

Comment:
```suggestion
        self._args = (float(a), float(b), float(c), float(d))
```

We apply a "fast fail" approach by immediately casting a, b, c, and d to floats. This is beneficial because sequences are often cast to Quaternion in the class to leverage its operations.